validator/api/src/main/java/tech/pegasys/teku/validator/api/UpdatableGraffitiProvider.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.api;
+
+import static tech.pegasys.teku.validator.api.GraffitiManager.GRAFFITI_MANAGEMENT_DIR;
+
+import java.nio.file.Path;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.service.serviceutils.layout.DataDirLayout;
+
+public class UpdatableGraffitiProvider implements GraffitiProvider {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Path graffitiPath;
+  private final GraffitiProvider defaultProvider;
+
+  public UpdatableGraffitiProvider(
+      final DataDirLayout dataDirLayout,
```

Comment:
One thing I do not like about this approach is the fact that two objects now need to know details about where the graffiti is being saved and the format of the file, etc.

I think centralizing this knowledge on the `GraffitiManager` is more cohesive.

Have you considered this approach?